File: VisitBosnia/VisitBosnia.Model/Attraction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace VisitBosnia.Model
{
    public partial class Attraction
    {
        [ForeignKey("IdNavigation")]
        public int Id { get; set; }
        //public string? AddressMap { get; set; }
        public decimal GeoLong { get; set; } 
        public decimal GeoLat { get; set; } 

        public virtual TouristFacility IdNavigation { get; set; } = null!;
    }
}

File: VisitBosnia/visit_bosnia_mobile/lib/model/attractions/attraction.dart
import 'package:visit_bosnia_mobile/model/tourist_facility.dart';

class Attraction {
  int? id;
  double? geoLong;
  double? geoLat;
  TouristFacility? idNavigation;

  Attraction({this.id, this.geoLong, this.geoLat, this.idNavigation});

  Attraction.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    geoLong = json['geoLong'];
    geoLat = json['geoLat'];
    idNavigation = json['idNavigation'] != null
        ? new TouristFacility.fromJson(json['idNavigation'])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['geoLong'] = this.geoLong;
    data['geoLat'] = this.geoLat;
    if (this.idNavigation != null) {
      data['idNavigation'] = this.idNavigation!.toJson();
    }
    return data;
  }
}
